src/helpers/Location.ts

```diff
@@ -5,30 +5,33 @@
  */
 
 export class Location extends URL {
-	constructor(url: string, base: string = document.baseURI) {
+	constructor(url: URL | string, base: string = document.baseURI) {
 		super(url.toString(), base);
 	}
 
-	get url() {
+	/**
+	 * The full local path including query params.
+	 */
+	get url(): string {
 		return this.pathname + this.search;
 	}
 
 	/**
 	 * Instantiate a Location from an element's href attribute
-	 * @param {Element} el
+	 * @param el
 	 * @return new Location instance
 	 */
-	static fromElement(el: HTMLAnchorElement): Location {
+	static fromElement(el: Element): Location {
 		const href = el.getAttribute('href') || el.getAttribute('xlink:href');
 		return new Location(href!);
```

Comment:
Maybe a drive-by improvement could be to make sure that the href is always a string here:
```js
const href = el.getAttribute('href') || el.getAttribute('xlink:href') || '';
return new Location(href);
```